### PR TITLE
Simplify usage of mapIds

### PR DESCRIPTION
**Pull request by mbeckem**
Based on user feedback: usage of explicit `mapId` is repetetive and inconvenient since most applications use a single map.

This PR introduces the `DefaultMapProvider` that allows you to configure a (default) map for an entire subtree of components.
In this case, the `mapId` parameter can be omitted, unless one wishes to use a different map (explicit map id on a component always overrides any inherited defaults).

This approach gets rid of the repetetiveness in the React UI part of an application (see updated example applications).

Example:

```tsx
import { DefaultMapProvider } from "@open-pioneer/map";

<DefaultMapProvider mapId={MAP_ID}>
    {/* no need to repeat the map id in this subtree, unless you want to use a different one */}
    <MapContainer />
    <Toc />
    <ComplexChild />
</DefaultMapProvider>;
```

This PR also implements support for explicit `map` props on all components (as an alternative to `mapId`).

### CODE REVIEW

**Review by antoniave — CHANGES_REQUESTED** [review]

File: src/packages/map/ui/MapContainer.tsx [hunk 1]
```diff
@@ -77,8 +73,8 @@ export function MapContainer(props: MapContainerProps) {
     const { containerProps } = useCommonComponentProps("map-container", props);
     const mapContainer = useRef<HTMLDivElement>(null);
     const mapAnchorsHost = useRef<HTMLDivElement>(null);
-    const modelState = useMapModel(mapId);
-    const mapModel = modelState.map;
+    const modelState = useMapModel(props);
+    const map = modelState.map;
```
[the remark this answers is not in the document]
Comment (mbeckem) [reply src/packages/map/ui/MapContainer.tsx:77]: Using `map` for an instance of `MapModel` is also done in the props of the new `DefaultMapProvider` and `useMapModel`:

```ts
export interface MapModelProps {
    /**
     * The id of the map.
     * The map will be looked up in the MapRegistry service.
     */
    mapId?: string | undefined;

    /**
     * The direct map model reference to use.
     * This property can be used as an alternative to the {@link mapId}.
     */
    map?: MapModel | undefined;
}
```

I believe `map` and `olMap` are different enough (i think there is also precedent for this naming in other packages).

File: src/packages/map/ui/useMapModel.ts [hunk 2]
```diff
@@ -35,24 +29,93 @@ export interface UseMapModelRejected {
...
  * The map model cannot be returned directly because it may not have completed its initialization yet.
  */
-export function useMapModel(mapId: string): UseMapModelResult {
+export function useMapModel(props: MapModelProps): UseMapModelResult;
+
+/**
+ * React hook that returns the default map model (if available, see {@link DefaultMapProvider}).
+ */
+export function useMapModel(): UseMapModelResult;
+export function useMapModel(props?: undefined | string | MapModelProps): UseMapModelResult {
```
[the remark this answers is not in the document]
Comment (mbeckem) [reply src/packages/map/ui/useMapModel.ts:75]: Are you sure? I've added a test that passes a mock map model directly to the hook. The hook throws an exception (the unexpected parameter is treated the same as an empty object). So passing garbage arguments to the hook will either return the default map model (if present in the component hierarchy) or thrown an error.

File: src/packages/map/ui/DefaultMapProvider.tsx [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+import { MapModelProps } from "./useMapModel";
+
+const DefaultMapContext = createContext<MapModelProps | undefined>(undefined);
+DefaultMapContext.displayName = "DefaultMapContext";
+
+/**
+ * Configures the given map as the default map for all child components.
+ * Child components do not need to specify the map explicitly, unless they wish to use a different one.
+ *
+ * Note that you can also use a direct reference to the map model as an alternative to the id.
```
[the remark this answers is not in the document]
Comment (mbeckem) [reply src/packages/map/ui/DefaultMapProvider.tsx:13]: Done, see incoming commit

File: src/packages/map/ui/useMapModel.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (antoniave) [reply src/packages/map/ui/useMapModel.ts:75]: The error is not thrown if an DefaultMapProvider is used.

Example: Update test with the following wrapper: 
```ts 
 wrapper: (props) => (
                <PackageContextProvider services={services} {...props}>
                    <DefaultMapProvider map={mapModel}>{props.children}</DefaultMapProvider>
                </PackageContextProvider>
            )
```

[the remark this answers is not in the document]
Comment (mbeckem) [reply src/packages/map/ui/useMapModel.ts:75]: True, but I don't think that we can catch that in general (arbitrary objects are allowed, as are objects without any of the "mapId" or "map" parameters). However, we could introduce a special case for _just_ instances of MapModel (via `instance of`) and return the map model directly.

[the remark this answers is not in the document]
Comment (mbeckem) [reply src/packages/map/ui/useMapModel.ts:76]: I've included the mentioned special case, see tests.

**Review by antoniave — APPROVED** [review]

**Pull request merged**